src/backend/mpi/swdsm.cpp

```diff
@@ -922,16 +931,13 @@ void argo_initialize(std::size_t argo_size, std::size_t cache_size){
 		pthread_barrier_init(&threadbarrier[i],NULL,i);
 	}
 
-	cachesize = 0;
-	if(cache_size > argo_size) {
-		cachesize += argo_size;
-	} else {
-		cachesize += cache_size;
-	}
-	cachesize += pagesize*CACHELINE;
+	/** Limit cache_size to at most argo_size */
+	cachesize = std::min(argo_size, cache_size);
+	/** Round the number of cache pages upwards */
+	cachesize = align_forwards(cachesize, pagesize*CACHELINE);
+	/** At least two pages are required to prevent endless eviction loops */
```

Comment:
not two pages, but two cachelines / pagelines / cache entries / cache entry slots?